pkg/client/cache/listers.go

```diff
@@ -106,6 +107,44 @@ func (s *StoreToNodeLister) List() (machines api.NodeList, err error) {
 	return machines, nil
 }
 
+func (s *StoreToNodeLister) NodeCondition(conditionType api.NodeConditionType, conditionStatus api.ConditionStatus) storeToNodeConditionLister {
```

Comment:
I had initially thought that as well, but it's awkward to filter this because it's a list of conditions and you only want the latest. It also might get a lot more complicated down the line (nodecontroller code is all written in way that indicates a lot more condition types/statuses that might influence each other down the line). Still put in a TODO?